### PR TITLE
Api 24834 section 7 service pay validation

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- Adds validation and error handling for form 526 v2 service pay attributes
-  Adds rspec test for added validations

## Related issue(s)
[API-24834](https://vajira.max.gov/browse/API-24834)

## Testing done

- Postman
- RSpec

## What areas of the site does it impact?
modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb
modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb
modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by rockwellwindsor-va — DISMISSED** [review]
Looks good to me.

Testing used - RSpec

File: modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb [hunk 1]
```diff
@@ -117,6 +119,54 @@ def missing_point_of_contact?
...
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: "If 'servicePay.futureMilitaryRetiredPay' is true, then " \
+                    "'servicePay.futureMilitaryRetiredPayExplanation' is required"
+          )
+        end
+      end
+
+      def validate_form_526_separation_pay!
+        validate_form_526_separation_pay_received_date!
+      end
```
Comment (stiehlrod) [inline modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb:157]: why not call this `validate_form_526_separation_pay_received_date!` method above, on line 126? Do we need another method just to call it?

File: modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json [hunk 2]
```diff
@@ -198,7 +198,7 @@
                 "retiredStatus": "RETIRED",
```
Comment (stiehlrod) [inline modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json:198]: Do we need to make changes to request.json, or example.json? Do we need to run `rake rswag:claims_api:build`?

Comment (mchristiansonVA) [reply modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json:198]: Updated example.json and added new swagger file.

**Review by rockwellwindsor-va — APPROVED** [review]
LGTM

**Pull request merged**